### ISSUE TITLE
Good for nothing but a segfault

### ISSUE THREAD

**Issue by flexoron** [trealla-prolog/trealla#610]
v2.57.24-9-gf484
```
$ cat peloid.pl
bog((mud,bog)).
% bog((mud,bog,pile(_))).
goal_expansion(pile(Bed),(Bed)) :- bog(Bed).
peat :- pile(_).
mud  :- write(' pulp').
bog  :- write(' bath').

$ tpl peloid.pl 
?- peat.
 pulp bath   true.
?- 
```
```
$ cat peloid.pl
% bog((mud,bog)).
bog((mud,bog,pile(_))). % the good-for-nothing  to reveal the segfault
: 
: 

$ tpl peloid.pl 
Error: max vars reached, peloid.pl:6
   throw(error(type_error(callable,dummy/0),dummy/0)).
Segmentation fault (core dumped)

**Comment by infradig** [comment]
Better than an infinite loop using all memory!

**Comment by flexoron** [comment]
```
if (Max == Moritz) {

switch(lang) {

case (rust) 
print "Sorry,Segmentation Failed not implemented yet but as Fault available soon(V2.0)"

case (java) 
(Exception,(Exception,Exception),(Exception,Exception,Exception) ...)
print "An Exception has occurred,an erro ha n excep n err no a eexrcrpoexerrex n has occult"

case (python) :
print "Please reformat process code, You can do this life with our pythonic psychotic pyro-debugger"

case (javascript) 
if (rust or java or python) then try rust or java or python

default:
print "crowning event happens"
operatic diva thunders praise and worship songs for rustproof,coffeeshops,pythonboots and javaskip
goto breed roses  

}
```
```
Error: syntax error, mismatched parens/brackets/braces
Ad:Nearby are:Visit: 'javanskipt coffeeshops' selling 'rustproof pythonboots'
```
